### PR TITLE
feat: add VercelModel adapter for Language Model Specification v3 providers

### PR DESCRIPTION
**Pull request by awsarron**
## Description

Adds `VercelModel`, an adapter that wraps any [Language Model Specification v3](https://github.com/vercel/ai/tree/main/packages/provider/src/language-model/v3) (`LanguageModelV3`) provider for use as a Strands model provider. This lets users bring models from the Vercel AI SDK ecosystem (e.g. `@ai-sdk/amazon-bedrock`, `@ai-sdk/openai`, `@ai-sdk/anthropic`, etc.) directly into Strands agents.

The adapter handles:
- Streaming text, reasoning, and tool use (both incremental `tool-input-*` and complete `tool-call` events)
- Message formatting: converts Strands messages (text, images, documents, video, tool use/results, reasoning) to the `LanguageModelV3Prompt` format
- Tool spec and tool choice mapping
- Usage/token tracking including cache read/write tokens
- Error classification: maps `APICallError` status codes and error message patterns to `ModelThrottledError`, `ContextWindowOverflowError`, and `ModelError`
- All `LanguageModelV3CallOptions` settings (temperature, topP, topK, penalties, stop sequences, seed) forwarded through config

Importable as a subpath export:
```typescript
import { VercelModel } from '@strands-agents/sdk/models/vercel'
```

Usage:

OpenAI
```typescript
import { Agent } from '@strands-agents/sdk'
import { VercelModel } from '@strands-agents/sdk/models/vercel'
import { openai } from '@ai-sdk/openai'

const agent = new Agent({
  model: new VercelModel({ model: openai('gpt-4o') }),
})

const result = await agent.invoke('Hello!')
```

Bedrock
```typescript
import { Agent } from '@strands-agents/sdk'
import { VercelModel } from '@strands-agents/sdk/models/vercel'
import { bedrock } from '@ai-sdk/amazon-bedrock'

const agent = new Agent({
  model: new VercelModel({
    model: bedrock('us.anthropic.claude-sonnet-4-20250514-v1:0'),
    temperature: 0.7,
  }),
})

const result = await agent.invoke('Hello!')
```

`@ai-sdk/provider` is added as an optional peer dependency since it crosses the API boundary (users construct `LanguageModelV3` instances and pass them in).

## Related Issues

strands-agents/sdk-typescript#214
strands-agents/sdk-typescript#540

## Documentation PR

https://github.com/strands-agents/docs/pull/689

## Type of Change

New feature

## Testing

- 710-line unit test suite covering text/reasoning/tool streaming, message formatting (user, assistant, system), tool spec/choice mapping, usage mapping, error classification, config forwarding, and edge cases
- Integration test fixtures added for `VercelModel` with both Bedrock and OpenAI backends

How have you tested the change?

- [x] I ran `npm run check`

## Checklist
- [x] I have read the CONTRIBUTING document
- [x] I have added any necessary tests that prove my fix is effective or my feature works
- [x] I have updated the documentation accordingly
- [x] I have added an appropriate example to the documentation to outline the feature, or no new docs are needed
- [x] My changes generate no new warnings
- [x] Any dependent changes have been merged and published

----

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/models/vercel.ts [hunk 1]
```diff
@@ -0,0 +1,568 @@
...
+  // Each tool result gets its own tool message to preserve ordering
+  for (const result of toolResults) {
+    prompt.push({ role: 'tool', content: [result] })
+  }
+}
+
+/**
+ * Formats tool result content to LanguageModelV3 ToolResultOutput.
+ */
+function formatToolResultOutput(
```
[the remark this answers is not in the document]
Comment (awsarron) [reply src/models/vercel.ts]: Updated

**Review by pgrayy — DISMISSED** [review]
